python/modules/map_plate.py

```diff
@@ -41,95 +111,131 @@ def _getting_paths_mp(input_path):
-    print(mp_deltimer)
-    f_paths_map_plate = _getting_paths_mp(path_map_plate)
-    f_paths_map_plate = sorted(f_paths_map_plate, key = str)
-    logger.debug("f_paths_map_plate %s.", f_paths_map_plate)
-    f_paths_output = _making_path_list(path_output, csv_names)
-    logger.debug("f_paths_output %s.", f_paths_output)
-    if isinstance(path_csv,str):
-        f_paths_csv = _making_path_list(path_csv, csv_names)
-        logger.debug("f_paths_csv %s.", f_paths_csv)
-    mp_output = _parsing_map_plate(path_map_plate, f_paths_map_plate, mp_deltimer, exp_part, exp_id, exp_parts_all)
-    logger.debug("mp_output %s.", mp_output)
-    names = _getting_column_titles(path_map_plate, f_paths_map_plate)
-    logger.debug("names %s.", names)
-    for i in range(len(f_paths_output)):
-        if isinstance(path_csv,dict):
-            in_data = paths_csv[csv_names[i]]  # return dictrionary
-        elif isinstance(path_csv,str):
-            in_data = CSV_M.read_csv(f_paths_csv[i], csv_deltimer) # write to file  
+def get_well_params(mp_dict, well):
+    """ 
+    Get all parameters values for a given well
+    """
+    values = []
+    for param in mp_dict[well]:
+        values.append(mp_dict[well][param])
+    return values
+
+def _verify_input_mp_file(input_file, delimiter, dimensions = 0):
+    if not FM.file_verify_extension(input_file, ".csv"):
+        logger.error("Wrong input format: %s", input_file)
+        return False
+    if dimensions != 0:
+        data = CSV_M.read_csv(input_file, delimiter)
+        x = len(data)
+        y = len(data[0])
+        if dimensions != [x,y]:
+            logger.error("Wrong input file "
+                        "(improper csv dimensions): %s.", input_file)
+            return False
+    return True
+
+
+def _prepare_output(input_data, mp_dict, compare_column):
+    output_data = []
+    column_names = input_data[0]
+    for i in range(len(column_names)):
+        if column_names[i] == compare_column:
+            position = i
+    new_line = column_names + get_params_names(mp_dict)
+    output_data.append(new_line)
+    for i in range(1, len(input_data)):
+        key = input_data[i][position]
+        new_line = input_data[i] + get_well_params(mp_dict, key)
+        output_data.append(new_line)
+    return output_data
+
+def apply_mp(input_path, output_path, delimiter, mp_dict, csv_names, compare_column):
+    """
+    Main function of TASK_APPLY_MAP_PLATE. 
+    Write all map_plate information (from mp_dictionary) 
+    into given csv files.
+    """
+    input_paths_list = FM.filenames_make_paths_list(input_path, csv_names)
+    logger.debug("Apply map_plate: CSV input files list: "
+                "%s.",input_paths_list)
+    output_paths_list = FM.filenames_make_paths_list(output_path, csv_names)
+    logger.debug("Apply map_plate: CSV output files list: "
+                "%s.", output_paths_list)
+    for i in range(len(input_paths_list)):
```

Comment:
Fixed. However, here, I decided to use zip(input_paths_list, output_paths_list), I belive this solution is good for that case.
Thanks!